### PR TITLE
[#2242] docs(open-api): Add the open-api for access control

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add the open-api for access control.

### Why are the changes needed?

Fix: #2242

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By hand.

### CODE REVIEW

**Comment by jerqi** [comment]
@mchades Could you help me review this pull request?

**Comment by mchades** [comment]
Overall LGTM.

> I tried to merge user_permissions.yaml and group_permissions.yaml. But it throws an error.

Can you describe the steps to reproduce it? Or provide detailed error information?

**Comment by jerqi** [comment]
> Overall LGTM.
> 
> > I tried to merge user_permissions.yaml and group_permissions.yaml. But it throws an error.
> 
> Can you describe the steps to reproduce it? Or provide detailed error information?

Put them into one file.

[1] ../docs/open-api/user_permissions.yaml:118:5 at #/paths/~1metalakes~1{metalake}~1permissions~1users~1{user}~1revoke/~1metalakes~1{metalake}~1permissions~1groups~1{group}~1grant

Property `/metalakes/{metalake}/permissions/groups/{group}/grant` is not expected here.

116 |       $ref: "./openapi.yaml#/components/responses/ServerErrorResponse"
117 |
118 | /metalakes/{metalake}/permissions/groups/{group}/grant:
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
119 |   parameters:
120 |     - $ref: "./openapi.yaml#/components/parameters/metalake"

referenced from ../docs/open-api/openapi.yaml:138:5 at #/paths/~1metalakes~1{metalake}~1permissions~1users~1{user}~1revoke 

Error was generated by the spec rule.


[2] ../docs/open-api/user_permissions.yaml:165:5 at #/paths/~1metalakes~1{metalake}~1permissions~1users~1{user}~1revoke/~1metalakes~1{metalake}~1permissions~1groups~1{group}~1revoke

Property `/metalakes/{metalake}/permissions/groups/{group}/revoke` is not expected here.

163 |         $ref: "./openapi.yaml#/components/responses/ServerErrorResponse"
164 |
165 | /metalakes/{metalake}/permissions/groups/{group}/revoke:
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
166 |   parameters:
167 |     - $ref: "./openapi.yaml#/components/parameters/metalake"

referenced from ../docs/open-api/openapi.yaml:138:5 at #/paths/~1metalakes~1{metalake}~1permissions~1users~1{user}~1revoke 

Error was generated by the spec rule.


[3] ../docs/open-api/openapi.yaml:141:5 at #/paths/~1metalakes~1{metalake}~1permissions~1groups~1{group}~1grant

Can't resolve $ref: ENOENT: no such file or directory '/Users/roryqi/Code/gravitino/docs/open-api/users_permissions.yaml'

139 |
140 | /metalakes/{metalake}/permissions/groups/{group}/grant:
141 |   $ref: "./users_permissions.yaml#/paths/~1metalakes~1%7Bmetalake%7D~1permissions~1groups~1%7Bgroup%7D~1grant"
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
142 |
143 | /metalakes/{metalake}/permissions/groups/{group}/revoke:

Error was generated by the no-unresolved-refs rule.


[4] ../docs/open-api/openapi.yaml:144:5 at #/paths/~1metalakes~1{metalake}~1permissions~1groups~1{group}~1revoke

Can't resolve $ref: ENOENT: no such file or directory '/Users/roryqi/Code/gravitino/docs/open-api/users_permissions.yaml'

142 |
143 |   /metalakes/{metalake}/permissions/groups/{group}/revoke:
144 |     $ref: "./users_permissions.yaml#/paths/~1metalakes~1%7Bmetalake%7D~1permissions~1groups~1%7Bgroup%7D~1revoke"
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
145 |
146 | components:

Error was generated by the no-unresolved-refs rule.

**Comment by mchades** [comment]
Indentation issue

File: docs/open-api/openapi.yaml [hunk 1]
```diff
@@ -109,6 +109,40 @@ paths:
...
+    $ref: "./owners.yaml#/paths/~1metalakes~1%7Bmetalake%7D~1owners~1%7BmetadataObjectType%7D~1%7BmetadataObjectFullName%7D"
+
+  /metalakes/{metalake}/permissions/users/{user}/grant:
+    $ref: "./user_permissions.yaml#/paths/~1metalakes~1%7Bmetalake%7D~1permissions~1users~1%7Buser%7D~1grant"
+
+  /metalakes/{metalake}/permissions/users/{user}/revoke:
+    $ref: "./user_permissions.yaml#/paths/~1metalakes~1%7Bmetalake%7D~1permissions~1users~1%7Buser%7D~1revoke"
+
+  /metalakes/{metalake}/permissions/groups/{group}/grant:
+    $ref: "./users_permissions.yaml#/paths/~1metalakes~1%7Bmetalake%7D~1permissions~1groups~1%7Bgroup%7D~1grant"
```
Comment (mchades) [inline docs/open-api/openapi.yaml:141]: users_permissions.yaml -> user_permissions.yaml

Comment (jerqi) [reply docs/open-api/openapi.yaml]: Fixed.

**Review by mchades — APPROVED** [review]
LGTM

**Pull request merged**